shark_turbine/kernel/wave/promotion.py

```diff
@@ -18,9 +18,11 @@ def apply_promotion_pattern(custom_node: Read | Write, allocate_node: Allocate):
             ).add_to_graph(custom_node.graph)
             custom_node.replace_all_uses_with(promoted_read)
             with custom_node.graph.inserting_before(promoted_read):
-                Write(
+                promoted_write = Write(
                     custom_node.fx_node, allocate_node.fx_node, elements_per_thread
                 ).add_to_graph(custom_node.graph)
+                custom_read = get_custom(promoted_read)
+                custom_read.write_dependency = promoted_write
```

Comment:
I assume this write_dependency will be used for barriers for shared memory read/writes and scheduling, but not sure if I see it currently being used aside from just setting the attribute?

Reply:
Yes, that's the PR I am going to submit next. This is required to get the expansion working since the expansion works by following the use-def chains and we have a break in the chain when we write to and then read from shared memory. Think of this as the write to shared memory returns a SSA value that is consumed by the read from shared memory.

Reply:
Ahhh, right right, we have that expansion thingy that backward slice DFS from final-writes. 

OK, so if we do not do this, the write to shared memory wouldn't be expanded?

Reply:
yes exactly. since it starts at the write node, it would propagate to read from shared memory but not higher.

Reply:
OK, makes sense, thanks for the explanation. :)